common/display/displayqueue.cpp

```diff
@@ -274,33 +275,26 @@ void DisplayQueue::HandleUpdateRequest(DisplayQueueItem& queue_item) {
 
   if (!display_plane_manager_->CommitFrame(current_composition_planes,
                                            pset.get(), flags)) {
-    succesful_commit = false;
-  } else {
-    display_plane_manager_->EndFrameUpdate();
-    previous_layers_.swap(queue_item.layers_);
-    previous_plane_state_.swap(current_composition_planes);
-  }
-
-  if (!succesful_commit || (flags & DRM_MODE_ATOMIC_ALLOW_MODESET))
     return;
+  }
 
+  display_plane_manager_->EndFrameUpdate();
 #ifdef DISABLE_EXPLICIT_SYNC
   compositor_.InsertFence(fence);
 #else
   if (fence > 0) {
-    compositor_.InsertFence(dup(fence));
-    fd_handler_.AddFd(fence);
-    out_fence_.Reset(fence);
+    if (previous_layers_.size() > 0) {
+      HWCPoll(fence, -1);
```

Comment:
With poll the flip becomes blocking now.
This will reduce the performance and 16ms delay added for every flip.
Instead we should add this release logic based on the page flip event notified from the kernel in the event thread